app/store/middleware.js

```diff
@@ -261,21 +247,22 @@ function cleanupState(action, keepCurrent = false) {
 
         if (post) {
             if (retentionPeriod && post.create_at < retentionPeriod) {
-                const postsInChannel = nextEntities.posts.postsInChannel[post.channel_id] || [];
-                const index = postsInChannel.indexOf(postId);
-                if (index !== -1) {
-                    postsInChannel.splice(index, 1);
-                }
+                // This post has been removed by data retention, so don't keep it
+                removeFromPostsInChannel(nextEntities.posts.postsInChannel, post.channel_id, postId);
+
                 return;
             }
 
+            // Keep the post
             nextEntities.posts.posts[postId] = post;
 
+            // And its reactions
             const reaction = payload.entities.posts.reactions[postId];
             if (reaction) {
                 nextEntities.posts.reactions[postId] = reaction;
             }
 
+            // And its files
```

Comment:
should we also keep the opengraph data for these posts?

Reply:
That's a good point. We *should* keep that, but there's no easy way to map that back to the post at this time. I'm not sure if there's anything else we can do to change that. Thankfully, we do have the backwards compatibility logic to reload that OpenGraph data, and this isn't a functional change from how it worked before anyway.

Reply:
yes I guess that is fine then